hi-ml-histopathology/src/histopathology/configs/classification/BaseMIL.py

```diff
@@ -138,7 +154,16 @@ def get_pooling_layer(self) -> Tuple[nn.Module, int]:
         num_features = num_encoding * self.pool_out_dim
         return pooling_layer, num_features
 
-    def create_model(self) -> DeepMILModule:
+    def get_outputs_handler(self) -> DeepMILOutputsHandler:
+        return DeepMILOutputsHandler(outputs_root=self.outputs_folder,
+                                     n_classes=self.data_module.train_dataset.N_CLASSES,
+                                     tile_size=self.tile_size,
+                                     level=1,
+                                     class_names=self.class_names,
+                                     primary_val_metric=MetricsKey.AUROC,
+                                     maximise=True)
+
+    def setup_model_creation(self) -> None:
```

Comment:
Suggestion: rename this method to `def get_model_encoder(self) -> TileEncoder:` and return `model_encoder` instead of setting it as an attribute. It's only used in `create_model()`, and this makes the method's purpose clear without relying on side-effects.